### ISSUE TITLE
Using ground truth information(target category) to change predicted item_scores in the TEST phase?

### ISSUE THREAD

**Issue by WangZesheng12** [resistzzz/HGCAN#1]
Excellent work! But I noticed it seems like you are using ground truth information, which is target category to modify the model-predicted item_scores in the TEST phase. Or it's just an ablation experiment?
<img width="1204" alt="iShot_2024-02-01_23 24 47" src="https://github.com/resistzzz/HGCAN/assets/45476404/ba9ed212-2782-4e7a-a2a1-1b9bce83378e">

**Comment by Opium1715** [comment]
> Excellent work! But I noticed it seems like you are using ground truth information, which is target category to modify the model-predicted item_scores in the TEST phase. Or it's just an ablation experiment? <img alt="iShot_2024-02-01_23 24 47" width="1204" src="https://private-user-images.githubusercontent.com/45476404/301555049-ba9ed212-2782-4e7a-a2a1-1b9bce83378e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MDM4OTEsIm5iZiI6MTcwOTcwMzU5MSwicGF0aCI6Ii80NTQ3NjQwNC8zMDE1NTUwNDktYmE5ZWQyMTItMjc4Mi00ZTdhLWEyYTEtMWI5YmNlODMzNzhlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA1Mzk1MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU1MzMzZTFiNzY3NTgzOGFmODBlMzdhNDFhOTg2NTBjZWQ1NTJjMDJhNDVhODg1OGZmODRjMjljY2U1ZGIyNzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WxbUzSlyDAugYhoiFMrLoJ5ZpwaFbIVTOaR6A-KcawI">

![image](https://github.com/resistzzz/HGCAN/assets/90845580/830e4bc2-8152-4d0d-afe2-31e73583294d)
It appears both in train and test phase. I think this conversion in train phase is a regularization for the train loss computation. I have add some code comments. And in test phase, it is just for computing loss function to display it. Hope it helps!

**Comment by Opium1715** [comment]
> Excellent work! But I noticed it seems like you are using ground truth information, which is target category to modify the model-predicted item_scores in the TEST phase. Or it's just an ablation experiment? <img alt="iShot_2024-02-01_23 24 47" width="1204" src="https://private-user-images.githubusercontent.com/45476404/301555049-ba9ed212-2782-4e7a-a2a1-1b9bce83378e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MTA2MTIsIm5iZiI6MTcwOTcxMDMxMiwicGF0aCI6Ii80NTQ3NjQwNC8zMDE1NTUwNDktYmE5ZWQyMTItMjc4Mi00ZTdhLWEyYTEtMWI5YmNlODMzNzhlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA2VDA3MzE1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTczMTQwYjIzOTgzNDFmNzhkOWZkZTM5YjkyMmFhNzY3MDEzYWE1MzY0NWVkMWI5MDZhNzg4YjNkOGI2MGQzMzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Dgqen5Iii1OZxX2Q7qxivOPOA1NlUNcoaHvKUMp83EI">

![image](https://github.com/resistzzz/HGCAN/assets/90845580/b320e8ee-a6d5-458a-94a1-f90a71dbc640)
It appears both in train and test phase. I think it is a regularization for the loss computation in train phase. And in test phase it also play the same role in it. As we can see, "dice_threshold" is a probability threshold for such regularization in the setting of hyper-parameters,  once the random dice is lower we do it, otherwise not. And we also could observed that the regularization is doing the opposite(ie. mask the correct answer, remain the false or irrelevant) for the loss computation.
Hope it helps you!
